### PR TITLE
chore: untangle protobuf/numpy test pinning

### PR DESCRIPTION
**Pull request by dongryeolle**
## What                                            
                                                     
  Two independent bits of dependency untangling toward supporting modern stacks
  (newer TensorFlow / numpy 2.x). Part of #2472.     
                                                                                                          
  **1. Drop the direct `protobuf>=3.20` runtime dependency.**
  `protobuf` is already a transitive dependency of `onnx`, `tensorflow` and
  `onnxruntime`, which agree on a compatible range. Declaring our own floor only                          
  risks resolver conflicts (e.g. blocking a newer TensorFlow that pulls a
  different protobuf — see #2409) without adding any safety.
                                                                                                          
  **2. Make the unit-test numpy constraint configurable instead of hardcoding `numpy<2`.**
  - `tests/utils/setup_test_env.sh` takes an optional 4th arg `numpy_spec`
    (default `"numpy<2"`).                                                                                
  - The `unit_test` composite action exposes a matching `numpy_spec` input
    (default `"numpy<2"`).                           
                                                                                                          
  Default behaviour is unchanged for every existing job — this only *enables* a                           
  future lane to run the suite under numpy 2.x.      
                                                                                                          
## Relationship to #2474                           
                                                                                                          
  This is the **packaging / CI** half of numpy-2 support and is complementary to
  the source-code half in #2474 (replacing numpy-2.0-removed aliases like
  `np.product` / `np.string_`). #2474 makes the code *run* under numpy 2.x; this                          
  PR makes it possible to *test* under numpy 2.x. There is no file overlap between
  the two. Both are Part of #2472.              
                                                                                                          
  ## Why not add the numpy>=2 lane here?                                                                  
                                                     
  numpy 2.x requires **TensorFlow >= 2.19** (the 2.13/2.15 wheels hard-require
  `numpy<2`), which also pulls in **Keras 3** — a combination the test matrix does                        
  not yet cover. Establishing that baseline is its own change; this PR is the                             
  enabler so it can land cleanly on top.             
                                                     
  ## Testing                                         
                                                     
  - `pyproject.toml` parses; `protobuf` removed from direct deps, `onnx` retained;                        
    protobuf still resolves transitively via onnx/tensorflow/onnxruntime.
  - `setup_test_env.sh` argument guard accepts 3–4 args and rejects 2/5; the
    `numpy_spec` default resolves to `numpy<2` when omitted and to the passed                             
    value otherwise.                                 
  - `.github/actions/unit_test/action.yml` is valid YAML and forwards the 4th arg.

### CODE REVIEW

File: tests/utils/setup_test_env.sh [hunk 1]
```diff
@@ -1,21 +1,25 @@
 #!/bin/bash
 
 # # Check if the argument is provided
-if [ "$#" -ne 3 ]; then
-    echo "Usage: $0 <tensorflow_version> <onnxruntime_version> <onnx_version>"
+if [ "$#" -lt 3 ] || [ "$#" -gt 4 ]; then
```
[the remark this answers is not in the document]
Comment (dongryeolle) [reply tests/utils/setup_test_env.sh:8]: Done — added `set -euo pipefail` so a failed pip install aborts the script                             
  instead of being masked. Verified the arg guard and the optional 4th arg still
  behave under set -u (default expansion is safe).

File: tests/utils/setup_test_env.sh [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
     exit 1
 fi
 
 # Assign the argument to a variable
 TF_VERSION=$1
 ORT_VERSION=$2
 ONNX_VERSION=$3
+# numpy constraint is configurable so a lane can exercise the suite under numpy
+# 2.x; default keeps the historical numpy<2 pin for the existing combinations.
+NUMPY_SPEC=${4:-"numpy<2"}
```
[the remark this answers is not in the document]
Comment (dongryeolle) [reply tests/utils/setup_test_env.sh]: Done — changed to NUMPY_SPEC="${4:-numpy<2}" (quoting the whole expansion).

**Review by andife — APPROVED** [review]
